src/Rules/Doctrine/ORM/PropertiesExtension.php

```diff
@@ -82,7 +83,15 @@ public function isInitialized(PropertyReflection $property, string $propertyName
 			return false;
 		}
 
-		return $metadata->isReadOnly && !$declaringClass->hasConstructor();
+		if ($declaringClass->hasConstructor()) {
+			return false;
+		}
+
+		if ($property instanceof PhpPropertyReflection && ($property->isReadOnly() || $property->isReadOnlyByPhpDoc())) {
```

Comment:
This logic seems wrong. It's not that `readonly` makes the property initialized, it's that `#[ORM\GeneratedValue]` / `@ORM\GeneratedValue` makes it initialized/always written.

You should look at `$metadata->generatorType` and `$metadata->getIdentifierFieldNames()` IMHO.